### PR TITLE
control-service: getting all jobs should now be consistent

### PR DESCRIPTION
**Pull request by mrMoZ1**
what: The control-service now uses LinkedHashMap when populating the graphQL filter and search criteria strategies.
Which ensures consistent filtering and sorting.

why: The previous usage of a HashMap would not guarantee any ordering of the key set which would lead to user issues when querying data from the API such as differently sorted lists of data jobs for the same query. This would cause more problems when we include pagination on a larger query - such as vdk list -a - users would see duplicated data, or missing data. Example of the same query with different ordering results can be seen below:

image

This was run on a local instance of the control-service with dummy data.

testing: Added unit test.

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java [hunk 1]
```diff
@@ -176,7 +182,13 @@ private Criteria<V2DataJob> populateCriteria(List<Filter> filterList) {
                     f -> f,
                     filter ->
                         strategyFactory.findStrategy(
-                            JobFieldStrategyBy.field(filter.getProperty()))));
+                            JobFieldStrategyBy.field(filter.getProperty())),
+                    (v1, v2) -> {
+                      throw new IllegalArgumentException(
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java:187]: What's the point of this exception? Won't it be thrown if there are duplicates?

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java:187]: This is the current default behavior of the toMap function we use. However when supplying a map provider we need to also provide a duplicate handling function:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-java.util.function.BinaryOperator-java.util.function.Supplier-

Current behavior:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-

Which states in the docs that it throws IllegalStateException on duplicates.
So to summarize the only thing that we are changing is the map type provider to LinkedHashMap provider.

**Review by mivanov1988 — APPROVED** [review]

**Review by dakodakov — APPROVED** [review]
